libs/ui-watt/src/lib/components/slider/watt-slider.component.ts

```diff
@@ -52,33 +56,86 @@ export class WattSliderComponent {
   /** Step between each value. */
   @Input() step = 1;
 
-  /** Manually define all steps. Overrules `min`, `max` and `step`. */
-  @Input() customSteps?: number[];
-
   /** The currently selected range value. */
   @Input() value: WattSliderValue = { min: this.min, max: this.max };
 
+  @ViewChild('maxRange') maxRange!: ElementRef<HTMLInputElement>;
+
+  @ViewChild('minRange') minRange!: ElementRef<HTMLInputElement>;
+
+  private _maxChangeSubscription!: Subscription;
+  private _minChangeSubscription!: Subscription;
+  private _providedStep = this.step;
   /**
    * Emits value whenever it changes.
    * @ignore
    */
   @Output() valueChange = new EventEmitter<WattSliderValue>();
 
-  /**
-   * @ignore
-   */
-  get options(): Options {
-    return {
-      floor: this.min,
-      ceil: this.max,
-      minRange: 1,
-      hideLimitLabels: true,
-      hidePointerLabels: true,
-      enforceStep: false,
-      enforceStepsArray: false,
-      step: this.step,
-      stepsArray: this.customSteps?.map((value) => ({ value })),
-    };
+  ngAfterViewInit(): void {
+    const maxRangeElement = this.maxRange.nativeElement;
+    const minRangeElement = this.minRange.nativeElement;
+    const maxChanged$ = fromEvent(maxRangeElement, 'change');
+    const minChanged$ = fromEvent(minRangeElement, 'change');
+
+    this.updateRange(this.value.min, this.value.max);
+
+    this._maxChangeSubscription = maxChanged$.subscribe((event) => {
+      const maxValue = (event.target as HTMLInputElement).valueAsNumber;
+      const minValue = minRangeElement.valueAsNumber || this.min;
+      this.updateRange(minValue, maxValue);
+
+      if (minValue <= maxValue) {
+        maxRangeElement.valueAsNumber = maxValue;
+      } else {
+        maxRangeElement.valueAsNumber = minValue;
+      }
+
+      // If the step is too large, we need to adjust it to fit the range.
+      const rest = this.max - maxValue;
+      if (rest < this.step) {
+        this.step = rest;
+      } else {
+        this.step = this._providedStep;
+      }
+
+      this.onChange({ min: minValue, max: maxValue });
+    });
+
+    this._minChangeSubscription = minChanged$.subscribe((event) => {
+      const minValue = (event.target as HTMLInputElement).valueAsNumber;
+      const maxValue = maxRangeElement.valueAsNumber || this.max;
+      this.updateRange(minValue, maxValue);
+
+      if (minValue > maxValue) {
+        minRangeElement.valueAsNumber = maxValue;
+      }
+
+      this.onChange({ min: minValue, max: maxValue });
+    });
+  }
+
+  ngOnDestroy(): void {
+    this._maxChangeSubscription.unsubscribe();
+    this._minChangeSubscription.unsubscribe();
+  }
+
+  private updateRange(minValue: number, maxValue: number) {
+    const rangeDistance = this.max - this.min;
+    const fromPosition = minValue - this.min;
+    const toPosition = maxValue - this.min;
+
+    const sliderColor = 'var(--watt-color-secondary-light)';
+    const rangeColor = 'var(--watt-color-primary)';
```

Comment:
nitpick: suggestion: we have a `WattColorHelperService` that can be used for getting color values instead of hard-coding them. It's worth having a look.